src/AutoUI/models/helpers.ts

```diff
@@ -204,15 +249,54 @@ export const generateSchemaFromRefScheme = (
 	};
 };
 
+export const getRefSchema = (schema: JSONSchema, refSchemePrefix: string) => {
+	const refScheme = parseDescriptionProperty(schema, 'x-ref-scheme') as
+		| string
+		| null;
+	return refScheme
+		? (get(schema, `${refSchemePrefix}${refScheme}`) as JSONSchema) ?? schema
+		: schema;
+};
+
+export const getPropertyScheme = (
+	schemaValue: JSONSchema | JSONSchemaDefinition,
+) => {
+	const json = isJSONSchema(schemaValue) ? parseDescription(schemaValue) : null;
+	return json && ('x-foreign-key-scheme' in json || 'x-ref-scheme' in json)
+		? json['x-foreign-key-scheme'] ?? json['x-ref-scheme']
+		: null;
+};
+
 export const getSubSchemaFromRefScheme = (
 	schema: JSONSchema | JSONSchemaDefinition,
 	refScheme?: string,
 ): JSONSchema => {
-	const referenceScheme = refScheme || sieve.getPropertyScheme(schema)?.[0];
+	const referenceScheme = refScheme || getPropertyScheme(schema)?.[0];
 	const convertedRefScheme = convertRefSchemeToSchemaPath(referenceScheme);
 	if (!convertedRefScheme) {
 		return schema as JSONSchema;
 	}
 	const properties = findInObject(schema, 'properties');
 	return get(properties, convertedRefScheme);
 };
+
+export const getSchemaFormat = (schema: JSONSchema) => {
+	const property = getSubSchemaFromRefScheme(schema);
+	const format = property.format ?? schema.format;
+	return format;
+};
+
+export const getSchemaTitle = (
+	jsonSchema: JSONSchema,
+	propertyKey: string,
+	refScheme?: string | undefined,
+) => {
+	if (!refScheme) {
+		return jsonSchema?.title || propertyKey;
+	}
+	return (
+		getSubSchemaFromRefScheme(jsonSchema, refScheme)?.title ??
```

Comment:
The return type says it shouldn't be possible to get a nullish value from `getSubSchemaFromRefScheme`
```suggestion
		getSubSchemaFromRefScheme(jsonSchema, refScheme).title ??
```